frontend/javascripts/admin/dataset/composition_wizard/04_configure_new_dataset.tsx

```diff
@@ -97,6 +140,35 @@ export function ConfigureNewDataset(props: WizardComponentProps) {
       return;
     }
 
+    let layersWithTransforms;
+    const { sourcePoints, targetPoints } = wizardContext;
+    try {
+      layersWithTransforms = withTransforms(layersWithoutTransforms, sourcePoints, targetPoints);
+    } catch (exception) {
+      const tryAugmentation = await new Promise((resolve) => {
+        Modal.confirm({
+          title: "Augment landmarks?",
+          content:
+            "The provided landmarks can't be used for affine estimation, possibly " +
+            "due to their planar nature. Should a constant translation in the Z " +
+            "direction be assumed, and the landmarks adjusted accordingly?",
+          onOk: () => resolve(true),
+          onCancel: () => resolve(false),
+        });
+      });
+      const augmentLandmarks = (points: Vector3[]) =>
+        points.concat(points.map((p) => [p[0], p[1], p[2] + 1]));
+      if (tryAugmentation) {
+        layersWithTransforms = withTransforms(
+          layersWithoutTransforms,
+          augmentLandmarks(sourcePoints),
+          augmentLandmarks(targetPoints),
+        );
+      } else {
+        throw exception;
+      }
+    }
```

Comment:
I am afraid, that always using z for "extrusion" won't cover all cases. In case I annotate all my point in the zy viewport without moving in x direction, the x axis value of all points should be equal. Thus, moving the points by z should fix the problem here. In case we want to catch all possible planes, we should calculate the otrhogol vector to the plane and move the nodes in that direction.


But I can also understand that this might be a bit too much for this pr and can be solved in a follow up pr :)